omx/gstomx.c

```diff
@@ -1384,6 +1387,7 @@ gst_omx_port_acquire_buffer (GstOMXPort * port, GstOMXBuffer ** buf)
     gst_omx_component_wait_message (comp, timeout == -2 ? GST_SECOND : timeout);
 
     /* And now check everything again and maybe get a buffer */
+    num_retries++;
```

Comment:
Having an extra attribute inside GstOmxPort an option to determine if we have to use the num_retries or not? That could prevent changing the acquire_buffer signature.

Reply:
It seems that GstOmxPort structure is more intended to exchange information about the component port settings, a change in there would spread through all the set_format functions of many elements available in this repository. The retry_limit is more a feature of the acquire_buffer  function itself, it modifies the internal flow of the acquire buffer logic as required.